includes/internal/emails/class-email-user-profile-settings.php

```diff
@@ -0,0 +1,232 @@
+		// phpcs:ignore WordPress.Security.ValidatedSanitizedInput.InputNotSanitized, WordPress.Security.ValidatedSanitizedInput.MissingUnslash, WordPress.Security.NonceVerification.Missing -- Nonce verified before hook is called, Input is sanitized in the next lines.
+		$email_subscriptions = $_POST['sensei-email-subscriptions'] ?? [];
+		if ( ! is_array( $email_subscriptions ) ) {
+			return;
+		}
+
+		$email_subscriptions = array_map( 'wp_unslash', $email_subscriptions );
+		$email_subscriptions = array_map( 'sanitize_text_field', $email_subscriptions );
+
+		$user_emails = $this->get_emails_for_user( $user_id );
+
+		foreach ( $user_emails as $identifier => $email ) {
+			$should_subscribe = in_array( $identifier, $email_subscriptions, true );
+
+			$should_subscribe
+				? $this->subscription->subscribe( $user_id, $identifier )
+				: $this->subscription->unsubscribe( $user_id, $identifier );
+		}
+	}
+
+	/**
+	 * Add email setting fields in user profile page.
+	 *
+	 * @internal
+	 *
+	 * @param WP_User $user The user object.
+	 */
+	public function maybe_add_email_settings( WP_User $user ): void {
+		$user_emails = $this->get_emails_for_user_grouped_by_type( $user->ID );
+
+		if ( empty( $user_emails ) ) {
+			return;
+		}
+
+		?>
+		<h3><?php esc_html_e( 'Sensei Email', 'sensei-lms' ); ?></h3>
+
+		<table class="form-table">
+			<?php
+			foreach ( $user_emails as $type => $emails ) {
+				?>
+					<tr>
+						<th scope="row">
+							<?php
+							switch ( $type ) {
+								case 'student':
+									esc_html_e( 'Student Emails', 'sensei-lms' );
+									break;
+								case 'teacher':
+									esc_html_e( 'Teacher Emails', 'sensei-lms' );
+									break;
+								default:
+									esc_html_e( 'Other Emails', 'sensei-lms' );
+									break;
+							}
+							?>
+						</th>
+						<td>
+							<?php
+							foreach ( $emails as $identifier => $email ) {
+								$this->render_email_subscription_setting( $user->ID, $identifier, $email );
+							}
+							?>
+						</td>
+					</tr>
+				<?php
+			}
+			?>
+		</table>
+		<?php
+	}
+
+	/**
+	 * Render email subscription setting.
+	 *
+	 * @internal
+	 *
+	 * @param int     $user_id    The user ID.
+	 * @param string  $identifier The email identifier.
+	 * @param WP_Post $email      The email post object.
+	 */
+	private function render_email_subscription_setting( int $user_id, string $identifier, WP_Post $email ): void {
+		$is_unsubscribed = ! $this->subscription->is_subscribed( $user_id, $identifier );
+		$description     = (string) get_post_meta( $email->ID, '_sensei_email_description', true );
+
+		?>
+		<fieldset>
+			<label for="<?php echo esc_attr( $this->get_field_id( $identifier ) ); ?>">
+				<input name="sensei-email-subscriptions[]" type="checkbox" value="<?php echo esc_attr( $identifier ); ?>" <?php checked( false, $is_unsubscribed ); ?> id="<?php echo esc_attr( $this->get_field_id( $identifier ) ); ?>">
+				<?php echo esc_html( $description ); ?>
+			</label>
+		</fieldset>
```

Comment:
Same here: could be extracted into a view file.

Reply:
Extracted here: https://github.com/Automattic/sensei/pull/7586/commits/d4bbe8184bd6442e36629c00764ba5228d998836